File: wirtualny_swiat/src/swiat/rosliny/WilczeJagody.java
package swiat.rosliny;

import swiat.Organizm;
import swiat.Swiat;

import java.awt.Color;

public class WilczeJagody extends Roslina {
    public WilczeJagody(final int polozenieX, final int polozenieY, final int wiek) {
        setNazwa("WilczeJagody");
        setSymbol('J');
        setSila(99);
        setInicjatywa(0);
        setWiek(wiek);
        setPolozenieX(polozenieX);
        setPolozenieY(polozenieY);
        setNiesmiertelnosc(false);
    }


    public WilczeJagody(final Organizm inny) {
        kopiujObiekt(inny);
    }


    @Override
    public Organizm stworzNowySklonowanyObiekt() {
        return new WilczeJagody(this);
    }


    @Override
    public void kolizja(Swiat swiat, Organizm atakujacy) {
        int polozenieX = getPolozenieX(), polozenieY = getPolozenieY(), polozenieXAtak = atakujacy.getPolozenieX(), polozenieYAtak = atakujacy.getPolozenieY();

        setCzyZyje(false);
        swiat.usunOrganizm(this, polozenieX, polozenieY);
        swiat.wypiszWiadomosc(atakujacy.getNazwa() + " zjada Wilcze Jagody" + wypiszPolozenie(polozenieX,polozenieY));

        if (!atakujacy.czyNiesmiertelny()) {
            atakujacy.setCzyZyje(false);
            swiat.usunOrganizm(atakujacy, polozenieXAtak, polozenieYAtak);
            swiat.wypiszWiadomosc("i umiera w wyniku ich zjedzenia");
        }
    }


    @Override
    public Color rysowanie() {
        return Color.blue;
    }
}

File: wirtualny_swiat_python/WilczeJagody.py
from Roslina import Roslina


class WilczeJagody(Roslina):
    def __init__(self, x, y, wiek):
        super().__init__(x, y, wiek)
        self.setNazwa("WilczeJagody")
        self.setSymbol('J')
        self.setSila(99)
        self.setInicjatywa(0)
        self.setNiesmiertelnosc(False)

    def stworzNowySklonowanyObiekt(self):
        return WilczeJagody(0, 0, 0)

    def kolizja(self, swiat, atakujacy):
        self.setCzyZyje(False)
        swiat.usunOrganizm(self, self.polozenieX, self.polozenieY)
        swiat.wypiszWiadomosc(atakujacy.getNazwa() + " zjada Wilcze Jagody"
                              + self.wypiszPolozenie(self.polozenieX, self.polozenieY))

        if atakujacy.czyNiesmiertelny() is False:
            atakujacy.setCzyZyje(False)
            swiat.usunOrganizm(atakujacy, atakujacy.getPolozenieX(), atakujacy.getPolozenieY())
            swiat.wypiszWiadomosc("i umiera w wyniku ich zjedzenia")

    def rysowanie(self):
        return "mediumblue"
